software/controller/lib/debugvars/vars.h

```diff
@@ -111,4 +111,30 @@ class Float : public Primitive32 {
   float value_;
 };
 
+template <size_t N>
+class FloatArray : public Base {
+ public:
+  FloatArray(const char *name, Access access, const char *units, const char *help = "",
+             const char *fmt = "%.3f")
+      : Base(Type::FloatArray, name, access, units, help, fmt) {}
+
+  FloatArray(const char *name, Access access, float initial_fill, const char *units,
+             const char *help = "", const char *fmt = "%.3f")
+      : Base(Type::FloatArray, name, access, units, help, fmt) {
+    data.fill(initial_fill);
+  }
+
+  FloatArray(const char *name, Access access, std::array<float, N> initial, const char *units,
+             const char *help = "", const char *fmt = "%.3f")
+      : Base(Type::FloatArray, name, access, units, help, fmt), data(initial) {}
+
+  void get_value(void *write_buff) override { std::memcpy(write_buff, data.data(), size()); }
+
+  void set_value(void *read_buf) override { std::memcpy(data.data(), read_buf, size()); }
+
+  size_t size() const override { return 4 * N; }
+
+  std::array<float, N> data;
```

Comment:
Also, this should be private...

Reply:
Interesting one. So, they say for struct members - no trailing underscore, for class members - yes, underscore. I had somehow interpreted this to mean that only private members have underscore, so if class has a public member it would be fall under struct rules. But their doc doesn't seem to view it that way.

However, I see the reason why I thought this: our clang-tidy static checker is configured as per this interpretation:
```
# We only require the "_" suffix for private and protected members in order to
# allow struct members (which are public) to not use the suffix
```

Not sure if there is a way to make the checker differentiate between classes and structs and treat those members differently. Fundamentally this is naming convention is either:
* redundant with what the compiler will check for anyways, member visibility outside scope, or
* functionally irrelevant - you can tell if class is a `struct` or a `class` by how variable is named, but those are almost the same anyways, except default visibility

I kind of prefer the way it is, a redundant but helpful visual cue rather than some piece of trivial information.
I will update our wiki to make this explicit.